contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -406,4 +459,68 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
   function isCapabilityDeprecated(bytes32 hashedCapabilityId) external view returns (bool) {
     return s_deprecatedHashedCapabilityIds.contains(hashedCapabilityId);
   }
+
+  /// @notice Adds a DON made up by a group of nodes that support a list
+  /// of capability configurations
+  /// @param nodes The nodes making up the DON
+  /// @param capabilityConfigurations The list of configurations for the
+  /// capabilities supported by the DON
+  /// @param isPublic True if the DON is public
+  function addDON(
+    bytes32[] calldata nodes,
+    CapabilityConfiguration[] calldata capabilityConfigurations,
+    bool isPublic
+  ) external onlyOwner {
+    uint32 id = s_donId;
+
+    s_dons[id].id = id;
+    s_dons[id].isPublic = isPublic;
+
+    for (uint256 i; i < capabilityConfigurations.length; ++i) {
+      CapabilityConfiguration calldata configuration = capabilityConfigurations[i];
+      bytes32 capabilityId = configuration.capabilityId;
+
+      if (!s_hashedCapabilityIds.contains(capabilityId)) revert CapabilityDoesNotExist(capabilityId);
+
+      if (s_dons[id].capabilityIds.contains(capabilityId)) revert DuplicateDONCapability(id, capabilityId);
+
+      s_dons[id].capabilityIds.add(capabilityId);
+      s_dons[id].capabilityConfigs[capabilityId] = configuration.config;
+    }
+
+    // TODO:  Verify that the nodes added support the relevant capability
```

Comment:
This is currently not trivial as the `Node` struct just has a `bytes32[]` field for the supported capability IDs.  We have 2 options to make this easier.

1.  We update this to an `EnumerableSet.Bytes32Set` so that we can check that a node supports a capability in O(1).  The downside with this is that we'd have to update the function signatures for `addNodes` and `updateNodes` as the `Node` struct will now have a nested mapping

2.  We define a separate `mapping(bytes32 nodeP2PId => EnumerableSet.Bytes32Set capabilityIds)` to avoid the problem above.  This does duplicate data though.

@DeividasK